src/utils/crypto.ts

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import CryptoJS from 'crypto-js';
+import { HASH_ITERATIONS, HASH_KEY_SIZE } from '../constants';
+import { IEncryptedData } from '../types';
+
+/**
+ * Hash a piece of information with the given options.
+ *
+ * @param {string} data Data to hash
+ * @param {{salt: string, iterations: number, pbkdf2Hasher: string}} [options={}] options for the hash algo
+ * @returns {{hash: string, salt: string, iterations: number, pbkdf2Hasher: string}}
+ */
+export function hashData(
+  data: string,
+  { salt, iterations = HASH_ITERATIONS, pbkdf2Hasher = 'sha1'}: {salt?: string, iterations?: number, pbkdf2Hasher?: string} = {},
```

Comment:
The old `wallet.js` also saved pbkdf2Hasher as sha1 on the storage but never used it.
I believed it was done to preserve for future use (if we want to change the hasher) so I kept it.

Reply:
I checked and we may change this, but the hasher argument is not a string with the name of the hash algo.
If we want to support different hashers we can have a map of `<hash algo name>: <hash algo impl>` so we can just pass `pbkdf2Hasher='sha256'` and it would work, what do you think?

Reply:
We can keep it like this, it's fine. But we should use this parameter in the method with the map

Reply:
Started using the `hasher` argument.
I made only sha1 (default) and sha256 supported but we can add more in the future.